### PR TITLE
v0.1.12.4 perf(dashboard): bounded-parallel job fan-out

### PR DESCRIPTION
**Pull request by DeliciousHouse**
## Summary
- New `lib/process-concurrent.ts` — order-preserving bounded-parallel map.
- `listMarketingReviewQueueForTenant` and `listMarketingCampaignsForTenant` now fan out across the per-tenant job list with concurrency=4 instead of `await`-in-a-loop.
- Expected wall-clock: 24-36s → single-digit seconds on tenants with 30+ jobs.
- 9 helper tests pin order preservation, concurrency cap, wall-clock speedup, error semantics, edge cases.

## Why
Live UI audit caught `/api/marketing/reviews` taking 24-36s in production, hanging the dashboard on "Loading…". Per-job work is dominated by 4 sequential awaits — there's no reason to serialize across jobs.

## Why concurrency=4
- `DB_POOL_MAX=20` default per container.
- `ARIES_WEB_CONCURRENCY=2`, so peak DB pressure = 4 × 2 = 8 connections.
- Well under the pool ceiling. CLAUDE.md guardrail #1 (don't blindly Promise.all DB chains) is satisfied.

## Test plan
- [x] `npm run verify` green (38 → 47 tests, +9 helper regressions)
- [x] `tsc --noEmit` clean
- [ ] Post-deploy: time the dashboard load on `aries.sugarandleather.com` and confirm `/api/marketing/reviews` is well under 10s.
- [ ] Post-deploy: confirm the campaigns list still renders in the correct order (newest first).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: lib/process-concurrent.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+ */
+export async function processConcurrent<T, U>(
+  items: readonly T[],
+  fn: (item: T, index: number) => Promise<U>,
+  concurrency: number,
+): Promise<U[]> {
+  if (items.length === 0) {
+    return [];
+  }
+  const safeConcurrency = Math.max(1, Math.min(Math.floor(concurrency), items.length));
```
[the remark this answers is not in the document]
Comment (DeliciousHouse) [reply lib/process-concurrent.ts]: Addressed in d684601: Number.isFinite guards against NaN/Infinity/-Infinity in concurrency. Two new tests pin NaN and +/-Infinity → safe clamped behavior.

File: lib/process-concurrent.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      }
+      try {
+        results[i] = await fn(items[i], i);
+      } catch (error) {
+        if (firstError === null) {
+          firstError = error;
+        }
+      }
+    }
+  }
```
[the remark this answers is not in the document]
Comment (DeliciousHouse) [reply lib/process-concurrent.ts:83]: Addressed in d684601: worker now checks firstError before claiming the next index, so new work stops while in-flight calls still settle. New test asserts <total items started when an early item rejects.

File: backend/marketing/runtime-views.ts [hunk 3]
```diff
@@ -1572,23 +1573,35 @@ export async function listMarketingCampaignsForTenant(
...
+    async (jobId) => {
+      const status = await getMarketingJobStatus(jobId);
+      if (status.tenantName === null && status.brandWebsiteUrl === null && status.status === 'error') {
+        return null;
+      }
+      const view = await buildCampaignWorkspaceView(jobId);
+      const pendingApprovals = (await buildReviewItemsForJob(jobId)).filter((item) => item.status !== 'approved').length;
+      const runtimeDoc = await loadMarketingJobRuntime(jobId);
+      const jobBrandKitExtractedAt = runtimeDoc?.brand_kit?.extracted_at ?? null;
+      return { status, view, pendingApprovals, jobBrandKitExtractedAt };
```
[the remark this answers is not in the document]
Comment (DeliciousHouse) [reply backend/marketing/runtime-views.ts]: Addressed in d684601: split listMarketingCampaignsForTenant into two phases. Phase 1 (status+view) -> dedup serially in input-order -> phase 2 (buildReviewItemsForJob+loadMarketingJobRuntime) only for survivors. Both phases bounded-parallel at concurrency=4.
